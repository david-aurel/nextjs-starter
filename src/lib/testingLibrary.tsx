import { render } from "@testing-library/react"
import React from "react"
import { ThemeProvider } from "styled-components"

import { theme } from "./theme/theme"

type RenderFunction = typeof render
const Providers: React.ComponentType = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

const customRender = ((ui, options) =>
  render(ui, { wrapper: Providers, ...options })) as RenderFunction

// eslint-disable-next-line import/export
export * from "@testing-library/react"
// eslint-disable-next-line import/export
export { customRender as render }
