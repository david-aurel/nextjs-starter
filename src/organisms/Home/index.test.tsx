import React from "react"

import { Home } from "."
import { render } from "../../lib/testingLibrary"

describe("Home", () => {
  it("renders", () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  })
})
