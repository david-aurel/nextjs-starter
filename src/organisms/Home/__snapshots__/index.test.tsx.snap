// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders 1`] = `
.c0 {
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
}

<div>
  <h1
    class="c0"
  >
    Hello World
  </h1>
</div>
`;
